gobblin-runtime/src/main/java/gobblin/runtime/api/GobblinInstanceDriver.java

```diff
@@ -22,8 +22,14 @@
  * {@link JobSpec}s and run them.
  * */
 @Alpha
-public interface GobblinInstanceDriver extends Service {
+public interface GobblinInstanceDriver extends Service, JobLifecycleListenersContainer {
   JobCatalog getJobCatalog();
+  /**
+   * Returns a mutable instance of the job catalog
+   * (if it implements the {@link MutableJobCatalog} interface). Implementation will throw
+   * ClassCastException if the current catalog is not mutable.
+   */
+  MutableJobCatalog getMutableJobCatalog();
```

Comment:
ClassCastException is a RuntimeException. It denotes an incorrect usage of the method, not a problem with the business logic.  Typically, unchecked exceptions are not declared because the calling code is not expected to recover from them. In this particular case, someone has instantiated the instance with a read-only catalog and they are trying to write to the catalog. The calling code cannot recover from this.

See also https://docs.oracle.com/javase/tutorial/essential/exceptions/runtime.html